### PR TITLE
Parses basic JSDoc function headers, plus code signature line.

### PR DESCRIPTION
**Pull request by JimGaleForce**

### CODE REVIEW

File: DocParser/JSDocFunction.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                        }
+                }
+            }
+
+            if (string.IsNullOrWhiteSpace(this.Name))
+            {
+                var fnLine = (items.PreviousLine != null && items.PreviousLine.Contains("function")) ? items.PreviousLine :
+                    (items.NextLine != null && items.NextLine.Contains("function")) ? items.NextLine : string.Empty;
+
+                if (!string.IsNullOrWhiteSpace(fnLine))
```
Comment (michael-hawker) [inline DocParser/JSDocFunction.cs:63]: Some quick comments at the top of the ifs for 'grab function name' and 'grab function parameters' would help knowing which parts of the parse function do what at a glance.

File: DocParser/JSDocHelper.cs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+            return results;
+        }
+
+        // Assumes text contains /** ... */ commenting. 'text' to contain a single comment block.
+        // Returns lines without comments
+        public static List<string> GetLines(string text)
+        {
+            var results = new List<string>();
+
+            var lines = text.Split(new[] { '\n' });
```
Comment (michael-hawker) [inline DocParser/JSDocHelper.cs:76]: Do you want to use Environment.Newline or you probably want to ensure you can handle both types of line endings in the same lib?  Probably good to handle mixed as well as you never know how clean the source input is for these cases.

File: DocParser/JSDocHelper.cs [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+                    iAt = iAt2;
+                }
+
+                iAt = text.IndexOf(left, iAt + 1);
+            }
+
+            return list.ToArray();
+        }
+
+        public static JSDocSplitSet[] SpecialSplit(string text, string left, string right)
```
Comment (michael-hawker) [inline DocParser/JSDocHelper.cs:128]: What makes this split special?  Full C# XML Comments for the helper methods could be useful.

File: DocParser/JSDocHelper.cs [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        }
+
+        public static string[] Split(string text, string left, string right)
+        {
+            var list = new List<string>();
+            var iAt = -1;
+            iAt = text.IndexOf(left, iAt+1);
+            while (iAt>-1)
+            {
+                var iAt2 = text.IndexOf(right, iAt+left.Length);
```
Comment (michael-hawker) [inline DocParser/JSDocHelper.cs]: the iAt and iAt2 you have everywhere are pretty similar to look at.  Would just 'I/j' or 'l/r' or 'leftIndex/rightIndex' be better?

File: DocParser/JSDocHelper.cs [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+                if (iAt2>-1)
+                {
+                    list.Add(text.Substring(iAt+left.Length, iAt2 - iAt - left.Length));
```
Comment (michael-hawker) [inline DocParser/JSDocHelper.cs]: inconsistent spacing between expressions, should have spaces around operators.  (not just here)

File: DocParser/JSDocFunction.cs (hunk 1 above)
Comment (JimGaleForce) [reply DocParser/JSDocFunction.cs:63]: yep - known undoc'd at the moment

File: DocParser/JSDocHelper.cs (hunk 2 above)
Comment (JimGaleForce) [reply DocParser/JSDocHelper.cs:76]: good point - mixed is best

File: DocParser/JSDocHelper.cs (hunk 3 above)
Comment (JimGaleForce) [reply DocParser/JSDocHelper.cs:128]: known - docs to come later (hence the original page 'quick docs' message)

**Pull request merged**